x/poolmanager/router.go

```diff
@@ -43,38 +43,11 @@ func (k Keeper) RouteExactAmountIn(
 			_outMinAmount = tokenOutMinAmount
 		}
 
-		// Get underlying pool type corresponding to the pool ID at the current routeStep.
-		swapModule, err := k.GetPoolModule(ctx, routeStep.PoolId)
-		if err != nil {
-			return osmomath.Int{}, err
-		}
-
-		// Execute the expected swap on the current routed pool
-		pool, poolErr := swapModule.GetPool(ctx, routeStep.PoolId)
-		if poolErr != nil {
-			return osmomath.Int{}, poolErr
-		}
-
-		// Check if pool has swaps enabled.
-		if !pool.IsActive(ctx) {
-			return osmomath.Int{}, types.InactivePoolError{PoolId: pool.GetId()}
-		}
-
-		spreadFactor := pool.GetSpreadFactor(ctx)
-
-		tokenInAfterSubTakerFee, err := k.chargeTakerFee(ctx, tokenIn, routeStep.TokenOutDenom, sender, true)
+		tokenOutAmount, err := k.SwapExactAmountIn(ctx, sender, routeStep.PoolId, tokenIn, routeStep.TokenOutDenom, _outMinAmount)
 		if err != nil {
 			return osmomath.Int{}, err
 		}
 
-		tokenOutAmount, err = swapModule.SwapExactAmountIn(ctx, sender, pool, tokenInAfterSubTakerFee, routeStep.TokenOutDenom, _outMinAmount, spreadFactor)
-		if err != nil {
-			return osmomath.Int{}, err
-		}
-
-		// Track volume for volume-splitting incentives
-		k.trackVolume(ctx, pool.GetId(), tokenIn)
-
```

Comment:
Note: made RouteExactAmountIn call SwapExactAmountIn so that we minimize the number of placed we need to make changes if we need to add low-level logic such as charging taker fee or tracking volume